### PR TITLE
Fix issue 15014 - Win64: Invalid C++ mangling for multiple long arguments

### PR DESCRIPTION
**Pull request by rainers**
The types with "longer" names that encode as 2 bytes are also referenced by number if used again. This is an incompatibility between dmc and cl.

https://issues.dlang.org/show_bug.cgi?id=15014

### CODE REVIEW

File: test/runnable/extra-files/cppb.cpp [hunk 1]
```diff
@@ -519,3 +519,35 @@ Visitor2* getVisitor2()
 {
     return &inst;
 }
+
+/******************************************/
+// issues detected by fuzzer
+#if _LP64
+#define longlong long
+#else
+#define longlong long long
```
Comment (yebblies) [inline test/runnable/extra-files/cppb.cpp]: Use `int64_t`...

Comment (rainers) [inline test/runnable/extra-files/cppb.cpp]: > Use int64_t...

Ok, done. The fuzzer could have written these to begin with ;-)
VC doesn't have stdint.h before VS2010, but that's probably no problem for the test suite.

Comment (rainers) [inline test/runnable/extra-files/cppb.cpp]: >> Use int64_t...
> 
> Ok, done.

Argh, doesn't work for Darwin/64.

Comment (rainers) [inline test/runnable/extra-files/cppb.cpp]: > Argh, doesn't work for Darwin/64.

Reverted back to `#define longlong`

Comment (yebblies) [inline test/runnable/extra-files/cppb.cpp]: I forgot about that.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**